File: buy_sell_stocks_ii.py
"""
Time Complexity : O(n) 
Space Complexity : O(1)
Did this code successfully run on Leetcode : Yes
Any problem you faced while coding this: none
"""

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if not prices or len(prices) == 0:
            return 0
        
        max_profit = 0
        
        for i in range(1, len(prices)):
            if prices[i] > prices[i-1]:
                max_profit += prices[i] - prices[i-1]
                
        return max_profit

File: buySellStocksII.java
// Time Complexity : O(n) where n is the number of elemetns in prices
// Space Complexity :  O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No

// Java Iterator interface reference:
// https://docs.oracle.com/javase/8/docs/api/java/util/Iterator.html
class Solution {
    public int maxProfit(int[] prices) {
      int profit  = 0;
      //sanity check
      if(prices == null || prices.length==0)
       
      {
        return profit;
      }
      
      for(int i = 0;i<prices.length-1;i++)
      {
        if(prices[i+1]>prices[i])
        {
          profit += prices[i+1] - prices[i];
        }
      }
        return profit;
    }
}